hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/client/OzoneQuota.java

```diff
@@ -19,52 +19,85 @@
-  public static final String OZONE_QUOTA_MB = "MB";
-  public static final String OZONE_QUOTA_GB = "GB";
-  public static final String OZONE_QUOTA_TB = "TB";
-
   /** Quota Units.*/
-  public enum Units { B, KB, MB, GB, TB }
+  public enum Units {
+    // the names and the ordering are important
+    B(1),
+    KB(OzoneConsts.KB),
+    MB(OzoneConsts.MB),
+    GB(OzoneConsts.GB),
+    TB(OzoneConsts.TB),
+    PB(OzoneConsts.PB),
+    EB(OzoneConsts.EB);
+
+    private final long size;
+    private final List<RawQuotaInBytes> cache;
+
+    Units(long size) {
+      this.size = size;
+      this.cache = createCache(this);
+    }
 
-  // Quota to decide how many buckets can be created.
-  private long quotaInNamespace;
-  // Quota to decide how many storage space will be used in bytes.
-  private long quotaInBytes;
-  private RawQuotaInBytes rawQuotaInBytes;
-  // Data class of Quota.
-  private static QuotaList quotaList;
+    private static List<RawQuotaInBytes> createCache(Units unit) {
+      final List<RawQuotaInBytes> quotas = new ArrayList<>(1024);
+      for (int i = 0; i < 1024; i++) {
+        quotas.add(new RawQuotaInBytes(unit, i));
+      }
+      return Collections.unmodifiableList(quotas);
+    }
+
+    public long getSize() {
+      return size;
+    }
 
-  /** Setting QuotaList parameters from large to small. */
+    RawQuotaInBytes getRawQuotaInBytes(long b) {
+      return b < cache.size() ? cache.get(Math.toIntExact(b))
+          : new RawQuotaInBytes(this, b);
+    }
+  }
+
+  private static final List<Units> PARSE_ORDER;
   static {
-    quotaList = new QuotaList();
-    quotaList.addQuotaList(OZONE_QUOTA_TB, Units.TB, TB);
-    quotaList.addQuotaList(OZONE_QUOTA_GB, Units.GB, GB);
-    quotaList.addQuotaList(OZONE_QUOTA_MB, Units.MB, MB);
-    quotaList.addQuotaList(OZONE_QUOTA_KB, Units.KB, KB);
-    quotaList.addQuotaList(OZONE_QUOTA_B, Units.B, 1L);
+    List<Units> reversed = new ArrayList<>(Arrays.asList(Units.values()));
+    Collections.reverse(reversed);
+    PARSE_ORDER = Collections.unmodifiableList(reversed);
   }
 
+  // Quota to decide how many buckets can be created.
+  private long quotaInNamespace;
+  // Quota to decide how many storage space will be used in bytes.
+  private final long quotaInBytes;
+  private final RawQuotaInBytes rawQuotaInBytes;
+
   /**
    * Used to convert user input values into bytes such as: 1MB-> 1048576.
    */
   private static class RawQuotaInBytes {
-    private Units unit;
-    private long size;
+    static RawQuotaInBytes valueOf(long quotaInBytes) {
+      Preconditions.assertTrue(quotaInBytes >= 0, () -> "quotaInBytes = " + quotaInBytes + " must be >= 0");
+      final int i = Long.numberOfTrailingZeros(quotaInBytes) / 10;
+      final Units unit = Units.values()[i];
+      final RawQuotaInBytes b = unit.getRawQuotaInBytes(quotaInBytes >> (i * 10));
+      Preconditions.assertSame(quotaInBytes, b.sizeInBytes(), "sizeInBytes");
+      return b;
+    }
```

Comment:
@adoroszlai , Just found that `valueOf(0)` will use EB.  (Fortunately, we added EB; otherwise, it becomes an ArrayIndexOutOfBoundsException).  Not a big deal, although we probably should fix it.
```java
  public static void main(String[] args) {
    final RawQuotaInBytes q = RawQuotaInBytes.valueOf(0);
    System.out.println("q = " + q);
  }
// q = 0 EB
```